velox/expression/VectorUdfTypeSystem.h

```diff
@@ -951,6 +956,83 @@ struct VectorWriter<Row<T...>> {
   size_t offset_ = 0;
 };
 
+template <typename... T>
+struct VectorWriter<RowWriterT<T...>> {
+  using vector_t = typename TypeToFlatVector<Row<T...>>::type;
+  using exec_out_t = typename VectorExec::resolver<RowWriterT<T...>>::out_type;
+
+  VectorWriter() {}
+
+  void init(vector_t& vector) {
+    rowVector_ = &vector;
+    // Ensure that children vectors are at least of same size as top row vector.
+    initVectorWritersInternal<0, T...>();
+    resizeVectorWritersInternal<0>(rowVector_->size());
+    writer_.initialize();
+  }
+
+  exec_out_t& current() {
+    return writer_;
+  }
+
+  vector_t& vector() {
+    return *rowVector_;
+  }
+
+  void ensureSize(size_t size) {
+    if (size > rowVector_->size()) {
+      rowVector_->resize(size, /*setNotNull*/ false);
+      resizeVectorWritersInternal<0>(rowVector_->size());
+    }
+  }
+
+  template <size_t N>
+  void resizeVectorWritersInternal(size_t size) {
```

Comment:
shouldn't this be private?  (at least the name suggests that and it's only used internally, externally ensureSize can be used)